spec/lang/well-formed.md

```diff
@@ -431,6 +431,28 @@ impl ValueExpr {
                         let isize_int = IntType { signed: Signed, size: T::PTR_SIZE };
                         Type::Int(isize_int)
                     }
+                    ConstructWidePointer(ptr_ty) => {
+                        let Type::Ptr(thin_ptr_ty) = left else {
+                            throw_ill_formed!("BinOp::ConstructWidePointer: invalid left type: not a pointer");
+                        };
+                        if thin_ptr_ty.meta_kind() != PointerMetaKind::None {
+                            throw_ill_formed!("BinOp::ConstructWidePointer: invalid left type: not a thin pointer");
+                        }
+
+                        // We could also allow constructing a thin pointer with the unit metadata here, rust does,
```

Comment:
If Rust allows this and we don't that means Minimize has to do more work. Also this seems inconsistent with `GetMetadata` supporting thin pointers. So I think I'd prefer to support thin pointers here.